File: 1260.cpp
#include<iostream>
#include<vector>
#include<queue>
#include<algorithm>

using namespace std;

vector<vector<int>> adjacencyList(1001, vector<int>());
vector<bool> visited(1001, false);

void DFS(int currentPosition){
	visited[currentPosition] = true;
	cout<<currentPosition<<" ";
	for(auto i : adjacencyList[currentPosition]){
		if(visited[i] != true){
			DFS(i);
		}
	}
}

void BFS(int currentPosition){
	queue<int> nextVertex;
	nextVertex.push(currentPosition);
	visited[currentPosition] = true;

	while(!nextVertex.empty()){
		int current = nextVertex.front();
		nextVertex.pop();
		cout<<current<<" ";
		for(auto i : adjacencyList[current]){
			if(visited[i] != true){
				visited[i] = true;
				nextVertex.push(i);
			}
		}
	}
}

int main(){
	int numOfVertex = 0;
	int numOfEdge = 0;
	int startingVertex = 0;

	cin>>numOfVertex>>numOfEdge>>startingVertex;
	visited[0] = true;

	for(int i = 0; i < numOfEdge; i++){
		int vertexI = 0;
		int vertexJ = 0;
		cin>>vertexI>>vertexJ;
		adjacencyList[vertexI].push_back(vertexJ);
		adjacencyList[vertexJ].push_back(vertexI);
	}
	
	for(int i = 1; i < numOfVertex + 1; i++){
		sort(adjacencyList[i].begin(), adjacencyList[i].end());
	}
	
	DFS(startingVertex);
	cout<<endl;

	for(int i = 0; i < 1001; i++){
		visited[i] = false;
	}

	BFS(startingVertex);
	cout<<endl;

	return 0;
}

File: 1260.swift
import Foundation

struct Queue<T> {
    var queue: [T] = []
    
    mutating func push(_ element: T) {
        self.queue.append(element)
    }
    
    mutating func pop() -> T?{
        if queue.isEmpty {
            return nil
        } else {
            let target = queue.first!
            self.queue.remove(at: 0)
            return target
        }
    }
    
    func isEmpty() -> Bool{
        return queue.isEmpty
    }
}

let inputs = readLine()!.split(separator: " ")
let numOfVertices = Int(inputs[0])!
let numOfEdges = Int(inputs[1])!
let startingVertex = Int(inputs[2])!

var adjacencyList: [[Int]] = Array(repeating: [Int](), count: 1001)
var visited: [Bool] = Array(repeating: false, count: 1001)

input()
DFS(startingVertex)
clear()
BFS()


func input() {
    for _ in 1...numOfEdges {
        let input = readLine()!.split(separator: " ")
        let from = Int(input[0])!
        let to = Int(input[1])!
        adjacencyList[from].append(to)
        adjacencyList[to].append(from)
    }
    for i in 1...numOfVertices {
        adjacencyList[i].sort(by: <)
    }
}

func clear() {
    print()
    for i  in 1...numOfVertices {
        visited[i] = false
    }
}

func DFS(_ current: Int) {
    print(current, terminator: " ")
    visited[current] = true
    for element in adjacencyList[current] {
        if visited[element] {
            continue
        }
        DFS(element)
    }
}

func BFS() {
    var queue = Queue<Int>()
    queue.push(startingVertex)
    while !queue.isEmpty() {
        if let popped = queue.pop(){
            if visited[popped] {
                continue
            }
            print(popped, terminator: " ")
            visited[popped] = true
            for element in adjacencyList[popped] {
                if visited[element] {
                    continue
                }
                queue.push(element)
            }
        }
    }
}

